Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/OptimisticDirectExecutionQueryBaselineTests.cs

```
@@ -667,6 +758,44 @@ private static async Task<IQueryPipelineStage> GetOdePipelineAsync(OptimisticDir
             return documentContainer;
         }
 
+        private static QueryPartitionProvider CreateCustomQueryPartitionProvider(string key, string value)
+        {
+            Dictionary<string, object> queryEngineConfiguration = new Dictionary<string, object>()
+            {
+                {"maxSqlQueryInputLength", 262144},
+                {"maxJoinsPerSqlQuery", 5},
+                {"maxLogicalAndPerSqlQuery", 2000},
+                {"maxLogicalOrPerSqlQuery", 2000},
+                {"maxUdfRefPerSqlQuery", 10},
+                {"maxInExpressionItemsCount", 16000},
+                {"queryMaxGroupByTableCellCount", 500000 },
+                {"queryMaxInMemorySortDocumentCount", 500},
+                {"maxQueryRequestTimeoutFraction", 0.90},
+                {"sqlAllowNonFiniteNumbers", false},
+                {"sqlAllowAggregateFunctions", true},
+                {"sqlAllowSubQuery", true},
+                {"sqlAllowScalarSubQuery", true},
+                {"allowNewKeywords", true},
+                {"sqlAllowLike", true},
+                {"sqlAllowGroupByClause", true},
+                {"maxSpatialQueryCells", 12},
+                {"spatialMaxGeometryPointCount", 256},
+                {"sqlDisableQueryILOptimization", false},
+                {"sqlDisableFilterPlanOptimization", false},
+                {"clientDisableOptimisticDirectExecution", false}
+            };
+
+            if (key != null && value != null)
```

I wouldn't do this check (and let the test fail if wrong values are being passed in). Alternatively, you can fail the test if any of these are false instead of silently continuing.